### PR TITLE
feat(pipe/release): Mark GitHub releases as non-draft only after all artifacts are uploaded.

### PR DESCRIPTION
**Pull request by nichtverstehen**
Previously end-users would see missing artifacts if trying to use latest version while artifacts are being uploaded.

This currently applies only to GitHub releases. GitLab does not support drafts, and I don't dare to make the change for Gitea since I don't use it (and can't test easily).

### CODE REVIEW

**Comment by nichtverstehen** [comment]
Hi! Any concerns with this change?

File: www/docs/customization/release.md [hunk 1]
```diff
@@ -289,6 +290,10 @@ release:
   # You can disable this pipe in order to not upload any artifacts.
   disable: true
 
+  # If set to true, will not auto-publish the release.
+  # Available only for GitHub and Gitea.
+  draft: true
+
```
Comment (caarlos0) [inline www/docs/customization/release.md:296]: this prop is already documented above

Comment (caarlos0) [inline www/docs/customization/release.md:296]:
```suggestion
```

**Review by caarlos0 — COMMENTED** [review]
one small comment, lgtm otherwise

thanks for the pr

File: www/docs/customization/release.md (hunk 1 above)
Comment (nichtverstehen) [reply www/docs/customization/release.md]: The duplicate piece above was under GitHub section, but perhaps repeating it for Gitea explicitly is not worth it. Applied your suggestion!

**Review by caarlos0 — APPROVED** [review]

**Comment by caarlos0** [comment]
refs https://github.com/goreleaser/goreleaser/pull/4744

**Comment by caarlos0** [comment]
refs https://github.com/goreleaser/goreleaser/pull/4745

**Pull request merged**